src/components/fx/hover.js

```diff
@@ -543,7 +548,7 @@ function _hover(gd, evt, subplot, noHoverEvent) {
         var thisSpikeDistance;
         for(var i = 0; i < pointsData.length; i++) {
             thisSpikeDistance = pointsData[i].spikeDistance;
-            if(thisSpikeDistance < minDistance && thisSpikeDistance <= spikedistance) {
+            if(thisSpikeDistance <= minDistance && thisSpikeDistance <= spikedistance) {
```

Comment:
@antoinerg Could you please explain why this change was required?

Reply:
I can't remember why so I reverted the change in 818223d6c245f4ebf091ff9ac6890ed31e4039bc ! Thanks @archmoj :male_detective:

Reply:
@antoinerg thanks for the revision.
Resolved.

Reply:
I think this equal sign was needed after all. 

When setting `spikedistance` to `-1` as we do for unified hover label, `thisSpikeDistance` is `Infinity` and `minDistance` is also `Infinity`. Without an equal sign, the condition is never met and we don't get spikelines. This can be seen in the first above Codepen: the spikeline does not appear anymore.

Reply:
This looks like a spike line bug.